include/ginkgo/core/matrix/batch_csr.hpp

```diff
@@ -274,6 +274,19 @@ class Csr final
     void scale(const array<value_type>& row_scale,
                const array<value_type>& col_scale);
 
+    /**
+     * Performs the operation this = alpha*I + beta*this.
+     *
+     * @param alpha the scalar for identity
+     * @param beta  the scalar to multiply this matrix
+     *
+     * @note Performs the operation in-place for this batch matrix
+     * @note This operation fails in case this matrix does not have all its
+     *       diagonal entries.
+     */
+    void add_scaled_identity(ptr_param<const MultiVector<value_type>> alpha,
```

Comment:
My first sight of `add_scaled_identity` is `this +=  scaled * I` but I do not have a good idea on it yet